addons/default/visiosoft/advs-module/src/Adv/AdvRepository.php

```diff
@@ -727,4 +727,17 @@ public function getNoImageClassifiedsReport()
 
         return $classifieds->paginate($limit, ['*'], 'page', $page);
     }
+
+    public function getClassifiedsByCoordinates($lat, $lng, $distance = 50)
+    {
+        return $this
+            ->currentAds()
+            ->whereNotNull('map_Val')
+            ->select(
+                DB::raw("*, ( 3959 * acos( cos( radians('$lat') ) * cos( radians( SUBSTRING_INDEX(map_Val, ',', 1) ) ) * cos( radians( SUBSTRING_INDEX(map_Val, ',', -1) ) - radians('$lng') ) + sin( radians('$lat') ) * sin( radians( SUBSTRING_INDEX(map_Val, ',', 1) ) ) ) ) AS distance")
```

Comment:
Where did you get the raw query ?

Reply:
Multiple resources from google

Reply:
Why distance is fix ? This is looks like havershine formula.

Reply:
Because I don't want to get all the ads since it will slow down the query, only ads around 50 will be enough I think. You can change it, it doesn't have to be 50 it could be anything, but I found 50 works well in this scenario